### DISCUSSION TITLE
body-parser deprecated undefined extended in Plus-API

### DISCUSSION THREAD

**Discussion by John21DE** [devlikeapro/waha#91] [Q&A]
Hi all.

I have this error when I start the whatsapp-http-api-plus:

`[Nest] 20  - 05/05/2023, 12:11:35 PM     LOG [InstanceLoader] AppModulePlus dependencies initialized +1ms
body-parser deprecated undefined extended: provide extended option dist/main.js:29:38
[Nest] 20  - 05/05/2023, 12:11:36 PM     LOG [RoutesResolver] SessionsController {/api/sessions}: +148ms
`

Does anyone know what causes this?
Obviously as a result I cannot see the swagger landing page on Port 3000.

**Comment by allburov** [comment]
Hi there! 
It's a warning, it doesn't affect the interface. 
We'll remove the warning it in the next release tho, thank you for the help!

--- 

The main problem with access usually occurs when using `-p 127.0.0.1:3000:3000/tcp` - it means "Do not accept any requests if they made not for `http://127.0.0.1`". To accept request outside the world - use `-p 3000:3000/tcp` instead
```bash
sudo docker run -it --rm -p 3000:3000/tcp --name whatsapp-http-api-plus -e WHATSAPP_SWAGGER_USERNAME=admin -e WHATSAPP_SWAGGER_PASSWORD=admin -e WHATSAPP_API_KEY=yoursecretkey devlikeapro/whatsapp-http-api-plus
```
We also **strongly** recommend using both admin password and api key - https://waha.devlike.pro/docs/how-to/security/
